src/Graph.ts

```diff
@@ -484,7 +484,7 @@ export default class Graph {
 				// filter out empty dependencies
 				for (let i = 0; i < chunkList.length; i++) {
 					const chunk = chunkList[i];
-					if (chunk.isEmpty && !chunk.entryModule) {
+					if (chunk.isEmpty && !chunk.entryModule && !chunk.isManualChunk) {
```

Comment:
i am wondering why not using .filter() here or .map() instead of:
```
 					for (const module of orderedModules) {
 						const chunkInstance = new Chunk(this, [module]);	 						const chunkInstance = new Chunk(this, [module]);
```

Reply:
I agree, the `.splice`/`i--` seems unnecessarily complicated to me while providing only limited performance benefits considering this loop is only run once.

Reply:
I'm really not sure what the suggestion is here. Can you please spell it out. Note that we are mutating an array while iterating it.

Reply:
```js
chunkList = chunkList.filter(chunk => !chunk.isEmpty || chunk.entryModule || chunk.isManualChunk);
```

Reply:
Reads a little nicer 😉

Reply:
Ah, thanks! One of those pieces of code you get so used to looking at you miss the simplification :)